hy/core/language.hy

```diff
@@ -481,8 +481,11 @@
 
 (defn xor [a b]
   "Perform exclusive or between two parameters"
-  (or (and a (not b))
-      (and b (not a))))
+  (if
+     (and (not a) (not b))  b
+     (and (not a)      b)   b
+     (and      a  (not b))  a
+                            False))
```

Comment:
`xor` has the same truth table as `or`, with one exception: `(and a b)`, and you want it to behave like `or` otherwise.